### PR TITLE
Otp 24 build

### PR DESCRIPTION
**Pull request by seanhinde**
This PR updates dependencies to otp-24 compatible versions and fixes the majority of build warnings for OTP-24

It also removes the parse transform aec_plugin which is superseded by the refactoring to allow for pluggable consensus modules.

An updated rebar3 executable is included which doesn't use erlang:stacktrace() internally.

This PR is funded by the ACF

### CODE REVIEW

File: apps/aechannel/test/aesc_fsm_SUITE.erl [hunk 1]
```diff
@@ -4937,7 +4937,7 @@ force_progress_triggers_slash(Cfg) ->
     wait_for_signed_transaction_in_block(dev1, SignedFP, Debug),
 
     _Tx = await_on_chain_report(I4, #{info => can_slash}, ?TIMEOUT),
-    _Tx = await_on_chain_report(R4, #{info => can_slash}, ?TIMEOUT),
+    _Tx1 = await_on_chain_report(R4, #{info => can_slash}, ?TIMEOUT),
```
Comment (uwiger) [inline apps/aechannel/test/aesc_fsm_SUITE.erl:4940]: Hmm, these _should_ actually be the same tx, so removing the '_' is perhaps more appropriate.

File: apps/aechannel/test/aesc_txs_SUITE.erl [hunk 2]
```diff
@@ -2783,7 +2783,7 @@ fp_use_onchain_oracle(Cfg) ->
 
                 % oracle query and answer
                 oracle_query(OQuestion, _ResponseTTL = 100),
-                oracle_response(OResponse, _ResponseTTL = 100),
+                oracle_response(OResponse, _ResponseTTL1 = 100),
```
Comment (uwiger) [inline apps/aechannel/test/aesc_txs_SUITE.erl:2786]: Or
```
oracle_query(OQuestion, ResponseTTL = 100),
oracle_response(OResponse, ResponseTTL),
```

File: apps/aechannel/test/aesc_utils_tests.erl [hunk 3]
```diff
@@ -262,8 +262,8 @@ deserialize_broken_close_solo_payload() ->
     {error, bad_offchain_state_type} =
         aesc_utils:check_solo_close_payload(ChannelPubKey,
                                             FromPubKey,
-                                            _Nonce = 1,
-                                            _Fee = 1,
+                                            _Nonce1 = 1,
```
Comment (uwiger) [inline apps/aechannel/test/aesc_utils_tests.erl:265]: Wouldn't it be nice to pass the parameters as map elements instead? (could be a separate PR)

File: apps/aehttp/test/aehttp_sc_SUITE.erl [hunk 4]
```diff
@@ -1730,7 +1730,7 @@ sc_ws_deposit_(Config, Origin, XOpts) when Origin =:= initiator
     {ok, _, #{<<"event">> := <<"deposit_locked">>}} = wait_for_channel_event(SenderConnPid, info, Config),
     {ok, _, #{<<"event">> := <<"deposit_locked">>}} = wait_for_channel_event(AckConnPid, info, Config),
     {ok, _, #{<<"state">> := _NewState}} = wait_for_channel_event(SenderConnPid, update, Config),
-    {ok, _, #{<<"state">> := _NewState}} = wait_for_channel_event(AckConnPid, update, Config),
+    {ok, _, #{<<"state">> := _NewState1}} = wait_for_channel_event(AckConnPid, update, Config),
```
Comment (uwiger) [inline apps/aehttp/test/aehttp_sc_SUITE.erl:1733]:
```suggestion
    {ok, _, #{<<"state">> := _}} = wait_for_channel_event(AckConnPid, update, Config),
```

File: apps/aehttp/test/aehttp_sc_SUITE.erl [hunk 5]
```diff
@@ -1810,7 +1810,7 @@ sc_ws_withdraw_(Config, Origin, XOpts) when Origin =:= initiator
 
     ct:log("withdraw_locked from both"),
     {ok, _, #{<<"state">> := _NewState}} = wait_for_channel_event(SenderConnPid, update, Config),
-    {ok, _, #{<<"state">> := _NewState}} = wait_for_channel_event(AckConnPid, update, Config),
+    {ok, _, #{<<"state">> := _NewState1}} = wait_for_channel_event(AckConnPid, update, Config),
```
Comment (uwiger) [inline apps/aehttp/test/aehttp_sc_SUITE.erl:1813]:
```suggestion
    {ok, _, #{<<"state">> := _}} = wait_for_channel_event(AckConnPid, update, Config),
```

File: apps/aehttp/test/aehttp_sc_SUITE.erl [hunk 6]
```diff
@@ -2530,7 +2530,7 @@ update_volley_(FirstPubkey, FirstConnPid, FirstPrivkey, SecondPubkey, SecondConn
                                                    <<"channels.update_ack">>,
                                                    Config),
     {ok, _, #{<<"state">> := _State}} = wait_for_channel_event(FirstConnPid, update, Config),
-    {ok, _,  #{<<"state">> := _State}} = wait_for_channel_event(SecondConnPid, update, Config),
+    {ok, _,  #{<<"state">> := _State1}} = wait_for_channel_event(SecondConnPid, update, Config),
```
Comment (uwiger) [inline apps/aehttp/test/aehttp_sc_SUITE.erl:2533]:
```suggestion
    {ok, _,  #{<<"state">> := _}} = wait_for_channel_event(SecondConnPid, update, Config),
```

**Review by uwiger — APPROVED** [review]

File: apps/aechannel/test/aesc_fsm_SUITE.erl (hunk 1 above)
Comment (seanhinde) [reply apps/aechannel/test/aesc_fsm_SUITE.erl:4940]: Nicely spotted. Will change

File: apps/aechannel/test/aesc_txs_SUITE.erl (hunk 2 above)
Comment (seanhinde) [reply apps/aechannel/test/aesc_txs_SUITE.erl:2786]: .. not sure I like the suggestion .. To do that kind of thing I'd rather move the assign onto its own line.

I suspect we will see more and more code like this with new shiny otp-24 warnings.

**Review by uwiger — APPROVED** [review]

**Review by velzevur — APPROVED** [review]
